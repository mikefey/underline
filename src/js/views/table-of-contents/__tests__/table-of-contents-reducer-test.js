import test from 'ava';
import {
  UPDATE_TABLE_OF_CONTENTS,
  SHOW_TABLE_OF_CONTENTS,
  HIDE_TABLE_OF_CONTENTS,
} from './../table-of-contents-constants';
import tableOfContentsReducer from './../table-of-contents-reducer.js';

const data = [
  { title: 'item 1', href: 'item-1.html' },
  { title: 'item 2', href: 'item-2.html' },
];

test('default state', (t) => {
  t.snapshot(tableOfContentsReducer(undefined, {}));
});

test('handles UPDATE_TABLE_OF_CONTENTS', (t) => {
  t.snapshot(tableOfContentsReducer(undefined, { type: UPDATE_TABLE_OF_CONTENTS, data }));
});

test('handles SHOW_TABLE_OF_CONTENTS', (t) => {
  t.snapshot(tableOfContentsReducer(undefined, { type: SHOW_TABLE_OF_CONTENTS }));
});

test('handles HIDE_TABLE_OF_CONTENTS', (t) => {
  t.snapshot(tableOfContentsReducer(undefined, { type: HIDE_TABLE_OF_CONTENTS }));
});
