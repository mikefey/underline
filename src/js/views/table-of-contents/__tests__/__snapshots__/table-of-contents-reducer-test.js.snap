exports[`default state 1`] = `"{\"clickedButtonIndex\":null,\"data\":null,\"show\":false}"`;

exports[`handles HIDE_TABLE_OF_CONTENTS 1`] = `"{"clickedButtonIndex":null,"data":null,"show":false}"`;

exports[`handles SHOW_TABLE_OF_CONTENTS 1`] = `"{\"clickedButtonIndex\":null,\"data\":null,\"show\":true}"`;

exports[`handles UPDATE_TABLE_OF_CONTENTS 1`] = `"{\"clickedButtonIndex\":null,\"data\":[{\"title\":\"item 1\",\"href\":\"item-1.html\"},{\"title\":\"item 2\",\"href\":\"item-2.html\"}],\"show\":false}"`;
